src/components/menu/MenuOpener/MenuOpenedByClick.tsx

```diff
@@ -0,0 +1,33 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+import { IMenuOpenedBySharedProps } from './MenuOpenedBySharedProps';
+import OutsideClickHandler from '../OutsideClick';
+import MenuPosition from '../Position/MenuPosition';
+
+const MenuOpenedByClick = (props: IMenuOpenedBySharedProps) => {
+    const hideMenu = () => props.onOpenedChange(false);
+    const toggleMenu = () => props.onOpenedChange(!props.opened);
+
+    const OutsideClickHandlerWrapped = (props: { children: React.ReactNode }) => (
+        <OutsideClickHandler onOutsideClick={hideMenu} useCapture={false}>
+            {props.children}
+        </OutsideClickHandler>
+    );
+
+    return (
+        <MenuPosition
+            opened={props.opened}
+            toggler={<div onClick={toggleMenu}>{props.toggler}</div>}
```

Comment:
please extract it somewhere so it's not hidden here in props

Reply:
Moved little higher.